asistente_ladm_col/gui/toolbar.py

```diff
@@ -113,35 +127,41 @@ def build_boundary(self, db):
                 build_boundaries_layer = self.geometry.build_boundaries(selected_boundaries_layer)
 
-                copy_boundary_layer = self.app.core.get_layer_copy(layers[db.names.LC_BOUNDARY_T])
-                boundaries_count = layers[db.names.LC_BOUNDARY_T].featureCount()
-                selected_boundaries_count = layers[db.names.LC_BOUNDARY_T].selectedFeatureCount()
                 build_boundaries_count = build_boundaries_layer.featureCount()
                 expected_boundaries_count = boundaries_count - selected_boundaries_count + build_boundaries_count
 
                 # Build boundaries should have generated at least one boundary.
-                if selected_boundaries_count > 0 and build_boundaries_count > 0:
+                if build_boundaries_count > 0:
                     with edit(layers[db.names.LC_BOUNDARY_T]):
                         # Delete selected features as they will be imported again from a newly created layer after processed
                         layers[db.names.LC_BOUNDARY_T].deleteSelectedFeatures()
 
                     # Bring back the features we deleted before, but this time, with the boundaries fixed
-                    self.app.core.run_etl_model_in_backgroud_mode(db, build_boundaries_layer, db.names.LC_BOUNDARY_T, False)
+                    self.app.core.run_etl_model_in_backgroud_mode(db, build_boundaries_layer, db.names.LC_BOUNDARY_T)
 
-                    # check if features were insert successful
+                    # check if features were inserted successfully
                     if layers[db.names.LC_BOUNDARY_T].featureCount() == expected_boundaries_count:
                         self.logger.info_msg(__name__, QCoreApplication.translate("ToolBar",
-                                                                                  "{} feature(s) was(were) analyzed generating {} boundary(ies)!").format(num_boundaries, build_boundaries_layer.featureCount()))
+                                                                                  "{} feature(s) was(were) analyzed generating {} boundary(ies)!").format(selected_boundaries_count, build_boundaries_layer.featureCount()))
 
                     else:
-                        # Clean layer because wrong data could be insert previously
-                        layers[db.names.LC_BOUNDARY_T].dataProvider().truncate()
+                        if layers[db.names.LC_BOUNDARY_T].featureCount() != boundaries_count - selected_boundaries_count:
+                            # Clean layer because wrong data could have been inserted previously
+                            expr = "{} IN ('{}')".format(db.names.T_ILI_TID_F, "','".join([str(t_ili_tid) for t_ili_tid in boundary_t_ili_tids]))
+                            layers[db.names.LC_BOUNDARY_T].selectByExpression(expr)
+                            layers[db.names.LC_BOUNDARY_T].invertSelection()
```

Comment:
If you need to select and then invert the selection, then the expression is wrong. 

Perhaps you need `expr = "{} NOT IN ('{}')"...`, don't you?

Reply:
Thanks for the observation, the selection is much more optimal. Adjustment made, available in commit 513f31a